### ISSUE TITLE
Wrong detection of constant base

### ISSUE THREAD

**Issue by kamilrakoczy** [chipsalliance/Surelog#2555]
TEST:
```
parameter int foo = 32'H7fFF;

module test;
endmodule
```
Surelog doesn't detect that constant is in hex, when ``'H`` is used. Using ``'h`` gives correct information.

**Comment by kamilrakoczy** [comment]
Same issue applies to other bases, as well as ``'S``, e.g.:
```
parameter int foo = 32'Sd1;

module test;
endmodule
```
gives: ``DEC:d1``, but ``32'sd1`` gives correct: ``DEC:1``.

**Comment by alaindargelas** [comment]
Fixed by #2585
